device/tt_silicon_driver.cpp

```diff
@@ -177,6 +183,9 @@ struct TTDeviceBase
     std::vector<DMAbuffer> dma_buffer_mappings;
 
     std::uint32_t read_checking_offset;
+
+    void* bar4_mapping = nullptr;
```

Comment:
Consider renaming to bar4_wc, and bar4_wc_size, to follow naming of other variables.